File: app/src/main/java/com/lazylee/lzywanandroid/data/entity/ProjectCategory.java
package com.lazylee.lzywanandroid.data.entity;

import android.os.Parcel;
import android.os.Parcelable;

public class ProjectCategory implements Parcelable {

    private int id;
    private int courseId;
    private String name;
    private int order;
    private int parentChapterId;
    private int visible;

    protected ProjectCategory(Parcel in) {
        id = in.readInt();
        courseId = in.readInt();
        name = in.readString();
        order = in.readInt();
        parentChapterId = in.readInt();
        visible = in.readInt();
    }

    public static final Creator<ProjectCategory> CREATOR = new Creator<ProjectCategory>() {
        @Override
        public ProjectCategory createFromParcel(Parcel in) {
            return new ProjectCategory(in);
        }

        @Override
        public ProjectCategory[] newArray(int size) {
            return new ProjectCategory[size];
        }
    };

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getCourseId() {
        return courseId;
    }

    public void setCourseId(int courseId) {
        this.courseId = courseId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getOrder() {
        return order;
    }

    public void setOrder(int order) {
        this.order = order;
    }

    public int getParentChapterId() {
        return parentChapterId;
    }

    public void setParentChapterId(int parentChapterId) {
        this.parentChapterId = parentChapterId;
    }

    public int getVisible() {
        return visible;
    }

    public void setVisible(int visible) {
        this.visible = visible;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(id);
        parcel.writeInt(courseId);
        parcel.writeString(name);
        parcel.writeInt(order);
        parcel.writeInt(parentChapterId);
        parcel.writeInt(visible);
    }

    @Override
    public String toString() {
        return "ProjectCategory{" +
                "id=" + id +
                ", courseId=" + courseId +
                ", name='" + name + '\'' +
                ", order=" + order +
                ", parentChapterId=" + parentChapterId +
                ", visible=" + visible +
                '}';
    }
}

File: app/src/main/java/com/lazylee/lzywanandroid/data/entity/ProjectCategory.kt
package com.lazylee.lzywanandroid.data.entity

import android.os.Parcel
import android.os.Parcelable

class ProjectCategory protected constructor(`in`: Parcel) : Parcelable {

    var id: Int = 0
    var courseId: Int = 0
    var name: String? = null
    var order: Int = 0
    var parentChapterId: Int = 0
    var visible: Int = 0

    init {
        id = `in`.readInt()
        courseId = `in`.readInt()
        name = `in`.readString()
        order = `in`.readInt()
        parentChapterId = `in`.readInt()
        visible = `in`.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(id)
        parcel.writeInt(courseId)
        parcel.writeString(name)
        parcel.writeInt(order)
        parcel.writeInt(parentChapterId)
        parcel.writeInt(visible)
    }

    override fun toString(): String {
        return "ProjectCategory{" +
                "id=" + id +
                ", courseId=" + courseId +
                ", name='" + name + '\''.toString() +
                ", order=" + order +
                ", parentChapterId=" + parentChapterId +
                ", visible=" + visible +
                '}'.toString()
    }

    companion object {

        val CREATOR: Parcelable.Creator<ProjectCategory> = object : Parcelable.Creator<ProjectCategory> {
            override fun createFromParcel(`in`: Parcel): ProjectCategory {
                return ProjectCategory(`in`)
            }

            override fun newArray(size: Int): Array<ProjectCategory> {
                return arrayOfNulls(size)
            }
        }
    }
}
